src/intrinsics/ecma262/ArrayPrototype.js

```diff
@@ -140,6 +141,23 @@ export default function(realm: Realm, obj: ObjectValue): void {
       // 1. Let O be ? ToObject(this value).
       let O = To.ToObject(realm, context);
 
+      // If we have an object that is an unknown array with numeric properties, then
+      // we can return a temporal here as we know nothing of the array's properties.
+      // This should be safe to do, as we never expose the internals of the array.
```

Comment:
Can we make assumptions about Array.prototype? It could have been modified in global code. Perhaps there should be a test that property copyWithin is a local data property?

Reply:
We have a pure scope check, which means that we're stating that the global hasn't been modified outside of Prepack's knowledge.

Also, sorry for asking a silly question, but what is a local data property and how do you envisage I might test it? I'm happy to, I'm just not sure what you mean. :)

Reply:
As things stand, the fact that Prepack might know which getter is being run when you evaluate O["copyWithin", does not make it OK to run this getter at runtime. At least not without further checks.

What I mean by local data property, is that "copyWithin" is known NOT to be local property of O, and is a also known to BE a local property of Array.prototype and it is furthermore known that Array.prototype.copyWithin is NOT a getter.

Reply:
I just pushed some changes to `get.js`, I believe it now addresses this concern. Let me know if I've missed something.

Reply:
Those changes look good, but how do they relate to this code? There is no check here that it is safe to call copyWithin.

Reply:
I've added checks to ensure all the prototype methods are not local methods. We can safely know they were on the prototype as all unknown arrays come from `get.js` only to the `NativeFunction` values in `ArrayPrototype.js`.